### PR TITLE
Various Formatter Improvements

### PR DESCRIPTION
**Pull request by DaelonSuzuka**
This introduces a new syntax for defining snapshot tests, and fixes at least #717, #734, #728.

Snapshot tests can now be defined in a single file using "blocks":

```gdscript
# --- IN ---
var  a  =   10
# --- OUT ---
var a = 10
```

I also FINALLY found a solution to remove leading empty newlines from blocks:

```gdscript
# --- IN ---
func test():

	pass
# --- OUT ---
func test():
	pass

# --- IN ---
class Test:

	func _ready():

		pass
# --- OUT ---
class Test:
	func _ready():
		pass
```

### CODE REVIEW

**Comment by limbonaut** [comment]
Breaks on lines in a multiline string:
```
func dump() -> String:
	return """
	{
		level_file: '%s',
		md5_hash: %s,
		text: '%s',
		level_size: %s,
		world_pos: %s,
		preview_size: %s,
		preview_pos: %s,
		preview_texture: %s,
		explorer_layer: %s,
		connections: %s,
	}
	""" % [level_file, md5_hash, text, level_size, world_pos, preview_size, preview_pos, preview_texture, explorer_layer, connections]
```

**Comment by DaelonSuzuka** [comment]
Thanks @limbonaut!

**Comment by limbonaut** [comment]
An issue with quotes and unique name specifier:
```gdscript
@onready var main_menu: Button = %"Main Menu"
```
👇
```gdscript
@onready var main_menu: Button = % "Main Menu"
```

**Comment by limbonaut** [comment]
That's all I found. Nice to see the old issues fixed, and it's shaping up to be a really solid formatter.

File: syntaxes/GDScript.tmLanguage.json [hunk 1]
```diff
@@ -274,6 +274,11 @@
 					"match": "[-]?([0-9][0-9_]*e[\\-\\+]?\\[0-9_])",
 					"name": "constant.numeric.float.gdscript"
 				},
+				{
+					"name": "constant.numeric.float.gdscript",
+					"match": "(?x)\n  (?<! \\w)(?:\n    (?:\n      \\.[0-9](?: _?[0-9] )*\n      |\n      [0-9](?: _?[0-9] )* \\. [0-9](?: _?[0-9] )*\n      |\n      [0-9](?: _?[0-9] )* \\.\n    ) (?: [eE][+-]?[0-9](?: _?[0-9] )* )?\n    |\n    [0-9](?: _?[0-9] )* (?: [eE][+-]?[0-9](?: _?[0-9] )* )\n  )([jJ])?\\b\n",
+					"captures": { "1": { "name": "storage.type.imaginary.number.gdscript" } }
```
Comment (AlfishSoftware) [inline syntaxes/GDScript.tmLanguage.json:280]: Imaginary number with `j` suffix? Are complex numbers supported syntax in GDScript now?
Is there a source proposal/docs or something?

Comment (DaelonSuzuka) [reply syntaxes/GDScript.tmLanguage.json:280]: Nope, I copied it from MagicPython and didn't notice the j's! Thanks for the catch.

Comment (AlfishSoftware) [reply syntaxes/GDScript.tmLanguage.json:280]: You probably want to double-check by testing whether it supports, e.g. lower and uppercase `e`, `E` and `f`, `F` suffixes. Also whether it supports `_` separators and `.` cases like `0.` or `.5`. Float literals syntax can vary a lot depending on the language, e.g. GDShader (or whatever it's going to be called) technically accepts even weird cases like `0.f` IIRC.

**Pull request merged**